pkg/migration/migrations/separated_intents.go

```diff
@@ -538,7 +538,18 @@ func postSeparatedIntentsMigration(
 			if bytes.HasPrefix(start, keys.TimeseriesPrefix) && bytes.HasPrefix(end, keys.TimeseriesPrefix) {
 				continue
 			}
-			if err := deps.DB.Migrate(ctx, start, end, cv.Version); err != nil {
+			// Try running Migrate on each range 5 times before failing the migration.
+			retries := 0
+			var err error
```

Comment:
we have a retry helper that does some things like check for context cancellation, I wonder if we should use that here?

Reply:
Good point - using that here instead. The exponential backoff is probably worth it too.